source/extensions/filters/http/jwt_authn/verifier.cc

```diff
@@ -367,6 +367,53 @@ JwtProviderList getAllProvidersAsList(const Protobuf::Map<std::string, JwtProvid
   return list;
 }
 
+class ExtractOnlyVerifierImpl : public BaseVerifierImpl {
+public:
+  ExtractOnlyVerifierImpl(const AuthFactory& factory, const JwtProviderList& providers,
+                          const BaseVerifierImpl* parent)
+      : BaseVerifierImpl(parent), auth_factory_(factory), extractor_(Extractor::create(providers)) {
+    ENVOY_LOG(info,
+              "JWT filter configured for claim extraction only - signature validation is disabled");
+  }
+
+  void verify(ContextSharedPtr context) const override {
+    ENVOY_LOG(debug, "Extracting JWT claims without signature validation");
+
+    auto& ctximpl = static_cast<ContextImpl&>(*context);
+    // Set allow_failed=true and allow_missing=true to bypass validation
+    // The key difference is we're telling the authenticator to extract claims
+    // even when signature validation would fail
+    auto auth =
+        auth_factory_.create(nullptr, absl::nullopt, true /* allow failed - this is critical */,
+                             true /* allow missing */);
+
+    extractor_->sanitizeHeaders(ctximpl.headers());
+    auth->verify(
```

Comment:
Do you still need to perform full `verify` here? If you only want to extract claims and forward them as headers, can you just do certain job only here (like addExtractedData).

I didn't take a very close look but my instinct is that there is something could be simplified here. That will be great if you can double check as you have looked into this code. Thanks!